Mathlib/Data/Matroid/Circuit.lean

```diff
@@ -166,4 +171,135 @@ lemma restrict_circuit_iff (hR : R ⊆ M.E := by aesop_mat) :
+    M.Circuit (M.fundCircuit e I) := by
+  apply (hI.inter_right _).insert_circuit_of_forall (by simp [heI])
+  · rw [(hI.subset _).closure_inter_eq_inter_closure, mem_inter_iff, and_iff_right hecl,
+      hI.closure_sInter_eq_biInter_closure_of_forall_subset _
+      (by simp (config := {contextual := true}))]
+    · simp
+    · exact ⟨I, rfl.subset, hecl⟩
+    exact union_subset rfl.subset (sInter_subset_of_mem ⟨rfl.subset, hecl⟩)
+  simp only [mem_inter_iff, mem_sInter, mem_setOf_eq, and_imp]
+  exact fun f hfI hf hecl ↦ (hf _ (diff_subset.trans inter_subset_left) hecl).2 rfl
+
+lemma Indep.mem_fundCircuit_iff (hI : M.Indep I) (hecl : e ∈ M.closure I) (heI : e ∉ I) :
+    x ∈ M.fundCircuit e I ↔ M.Indep (insert e I \ {x}) := by
+  obtain (rfl | hne) := eq_or_ne x e
+  · simp [hI.diff, mem_fundCircuit]
+  suffices (∀ t ⊆ I, e ∈ M.closure t → x ∈ t) ↔ e ∉ M.closure (I \ {x}) by
+    simpa [fundCircuit_eq_sInter hecl, hne, ← insert_diff_singleton_comm hne.symm,
+      (hI.diff _).insert_indep_iff, mem_ground_of_mem_closure hecl, heI]
+  refine ⟨fun h hecl ↦ (h _ diff_subset hecl).2 rfl, fun h J hJ heJ ↦ by_contra fun hxJ ↦ h ?_⟩
+  exact M.closure_subset_closure (subset_diff_singleton hJ hxJ) heJ
+
+lemma Base.fundCircuit_circuit {B : Set α} (hB : M.Base B) (hxE : x ∈ M.E) (hxB : x ∉ B) :
+    M.Circuit (M.fundCircuit x B) :=
+  hB.indep.fundCircuit_circuit (by rwa [hB.closure_eq]) hxB
+
+/-- For `I` independent, `M.fundCircuit e I` is the only circuit contained in `{e} ∪ I`. -/
+lemma Circuit.eq_fundCircuit_of_subset (hC : M.Circuit C) (hI : M.Indep I) (hCss : C ⊆ insert e I) :
+    C = M.fundCircuit e I := by
+  obtain hCI | ⟨heC, hCeI⟩ := subset_insert_iff.1 hCss
+  · exact (hC.not_indep (hI.subset hCI)).elim
+  suffices hss : M.fundCircuit e I ⊆ C by
+    refine Eq.symm <| (hI.fundCircuit_circuit ?_ (fun heI ↦ ?_)).eq_of_subset_circuit hC hss
+    · rw [hI.mem_closure_iff]
+      exact .inl (hC.dep.superset hCss (insert_subset (hC.subset_ground heC) hI.subset_ground))
+    exact hC.not_indep (hI.subset (hCss.trans (by simp [heI])))
+  refine insert_subset heC (inter_subset_right.trans ?_)
+  refine (sInter_subset_of_mem (t := C \ {e}) ?_).trans diff_subset
+  simp [hCss, hC.mem_closure_diff_singleton_of_mem heC]
+
+/-! ### Dependence -/
+
+lemma Dep.exists_circuit_subset (hX : M.Dep X) : ∃ C, C ⊆ X ∧ M.Circuit C := by
+  obtain ⟨I, hI⟩ := M.exists_basis X
+  obtain ⟨e, heX, heI⟩ := exists_of_ssubset
+    (hI.subset.ssubset_of_ne (by rintro rfl; exact hI.indep.not_dep hX))
+  exact ⟨M.fundCircuit e I, (M.fundCircuit_subset_insert e I).trans (insert_subset heX hI.subset),
+    hI.indep.fundCircuit_circuit (hI.subset_closure heX) heI⟩
+
+lemma dep_iff_superset_circuit (hX : X ⊆ M.E := by aesop_mat) :
+    M.Dep X ↔ ∃ C, C ⊆ X ∧ M.Circuit C :=
+  ⟨Dep.exists_circuit_subset, fun ⟨C, hCX, hC⟩ ↦ hC.dep.superset hCX⟩
+
+lemma dep_iff_superset_circuit' : M.Dep X ↔ (∃ C, C ⊆ X ∧ M.Circuit C) ∧ X ⊆ M.E :=
```

Comment:
Lemmas with a primed name like this and the following should have a docstring, see the discussion here: https://leanprover.zulipchat.com/#narrow/channel/287929-mathlib4/topic/theorem.20names.20with.20primes.3F/near/471505697. I'm surprised this case isn't causing the linter to fail, actually...